flopy/discretization/modeltime.py

```diff
@@ -7,61 +12,211 @@ class ModelTime:
-        self._start_datetime = start_datetime
+        nrecs = len(perlen)
+        if tsmult is None:
+            tsmult = np.full((nrecs,), 1)
+
+        if nrecs != len(nstp):
+            raise AssertionError()
+
+        if len(tsmult) != len(nstp):
+            raise AssertionError
+
+        self._perlen = perlen
+        self._nstp = nstp
+        self._tsmult = tsmult
+        self._time_units = self.timeunits_from_user_input(time_units)
+        self._start_datetime = self.datetime_from_user_input(start_datetime)
         self._steady_state = steady_state
+        self._totim_dict = {}
+        self.__str_format = "%Y-%m-%dt%H:%M:%S"
 
     @property
     def time_units(self):
+        """
+        Returns a normalized string representation of the time units
+        """
         return self._time_units
 
     @property
     def start_datetime(self):
+        """
+        Returns a datetime.datetime object of the model start time
+        """
         return self._start_datetime
 
     @property
     def perlen(self):
-        return self._period_data["perlen"]
+        """
+        Returns a list or array of stress-period lengths
+
+        """
+        return self._perlen.copy()
 
     @property
     def nper(self):
-        return len(self._period_data["perlen"])
+        """
+        Returns the number of stress periods
+        """
+        return len(self._perlen)
 
     @property
     def nstp(self):
-        return self._period_data["nstp"]
+        """
+        Returns a list or array of number of time steps per stress period
+        """
+        return self._nstp.copy()
 
     @property
     def tsmult(self):
-        return self._period_data["tsmult"]
+        """
+        Returns the time step multiplier value for each stress period
+
+        """
+        return self._tsmult.copy()
+
+    @property
+    def period_data(self):
+        """
+        Returns a tuple of period data for the MF6 TDIS package containing records
+        of [(perlen, nstp, tsmult), ....] for each stress period
+        """
+        return [(per, self.nstp[ix], self.tsmult[ix]) for ix, per in self.perlen]
 
     @property
     def steady_state(self):
+        """
+        Returns a boolean that indicates either steady-state or transient stress period
+
+        """
         return self._steady_state
 
     @property
     def totim(self):
+        """
+        Returns a list of totim values at the end of each time-step
+
+        """
+        if not self._totim_dict:
+            self._set_totim_dict()
+
+        return list(self._totim_dict.values())
+
+    @property
+    def kper_kstp(self):
+        """
+        Returns a list of kper, kstp tuples that correspond to totim
+
+        """
+        if not self._totim_dict:
+            self._set_totim_dict()
+        return list(self._totim_dict.keys())
+
+    @property
+    def tslen(self):
+        """
+        Method to get a list of time step lengths for the entire model period
```

Comment:
```suggestion
        Method to get a list of time step lengths for all time steps
```

to be consistent with above?

Reply:
updated